### DISCUSSION TITLE
Improve the Android experience.

### DISCUSSION THREAD

**Discussion by Booklight12** [godotengine/godot-proposals#6773] [Platforms]
Since godot was ported to Android, the development team has been working to improve the experience of using the Android engine. While the experience has greatly improved since the original version, and has benefited most of the student developer community, there is now a huge problem: how do we share the apps we've worked so hard to make? Currently, the related Settings exported by Android are deliberately hidden, which becomes a huge barrier to sharing. It's also discouraging for our particular group, and as an engine, I think the "export" function is one of the important features it has, and it's part of the experience. So far, most of the official replies to "export on Android" are "Android does not have the necessary conditions for export such as sdk". But as far as I know, now a number of components including Android sdk and ndk have corresponding implementations on Android. There are even applications that use them (e.g. AIDE: an Android integrated development environment that runs on the Android side and can export JAVA or Cpp into an apk that can be installed). Therefore, I think the lack of sdk and ndk should not be the reason to cut down the Android export function. At the same time, as a part of the use experience, I also hope that we can share the fruits of our labor more easily, rather than let the other side to carry out several tedious steps and finally run my "app" in the debug mode of the engine like me.

**Comment by Calinou** [comment]
Thanks for the proposal! Consolidating in https://github.com/godotengine/godot-proposals/discussions/4324.
